bolt-modules/boltlib/lib/puppet/functions/resolve_references.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+
+# Evaulates all _plugin references in a hash and returns the resolved reference data.
+Puppet::Functions.create_function(:resolve_references) do
+  # Resolve references.
+  # @param references A hash of reference data to resolve.
+  # @return A hash of resolved reference data.
+  # @example Resolve a hash of reference data
+  #   $references = {
+  #     "targets" => [
+  #       "_plugin" => "terraform",
+  #       "dir" => "path/to/terraform/project",
+  #       "resource_type" => "aws_instance.web",
+  #       "uri" => "public_ip"
+  #     ]
+  #   }
+  #
+  #   resolve_references($references)
+  dispatch :resolve_references do
+    param 'Hash[String[1], Any]', :references
+    return_type 'Hash[String[1], Any]'
+  end
+
+  def resolve_references(references)
+    unless Puppet[:tasks]
+      raise Puppet::ParseErrorWithIssue
+        .from_issue_and_stack(Bolt::PAL::Issues::PLAN_OPERATION_NOT_SUPPORTED_WHEN_COMPILING, action: 'resolve_references')
+    end
+
+    references = references.merge('name' => 'all') unless references['name']
+    inventory = Puppet.lookup(:bolt_inventory)
+    group = Bolt::Inventory::Group2.new(references, inventory.plugins)
+
+    group.resolve_references(references)
```

Comment:
This should maybe be a method on `Bolt::Plugin` instead of `Group2`. It's a bit strange to have to create a `Group2` instance for this...